.github/workflows/main.yml

```diff
@@ -34,7 +34,7 @@ jobs:
       fail-fast: false
       matrix:
         distribution: [ temurin ]
-        java_version: [ 8, 11, 16 ]
+        java_version: [ 8, 11, 17 ]
         scala_version: [ 2.12.15 ]
```

Comment:
Yep, we already depend on 2.12.15 and that ends up in the maven pom dependency list. So the only way this could really be an issue is if someone manually overrides that dependency. I wouldn't expect that to be common, but I actually recently came across a project that did just that, so it's not unheard of.

I guess errors are also possible if someone tries to use a previous Daffodil release on Java 17, since we don't test old releases. But there's nothing we can do about that.

Really, I think this is just something to be aware of, that dependency versions really do matter, even sometimes with patch releases. And often times, like in the case of using Scala 2.12.14 with Java 17, the reported exception was not helpful at all. The bug is here: [scala/bug/issues/12419](https://togithub.com/scala/bug/issues/12419), you can see the backtrace is all internal stuff with no obvious solution.